### PR TITLE
refactor certain promises into async/await

### PR DESCRIPTION
**Pull request by Levilian**
#2987

### CODE REVIEW

File: frontend/src/components/worksheets/BundleDetail/BundleDetail.jsx [hunk 1]
```diff
@@ -174,9 +174,8 @@ class BundleDetail extends React.Component<
                         }
                     }.bind(this),
                 );
-                $.when.apply($, fetchRequests).then(() => {
-                    this.setState(stateUpdate);
-                });
+                $.when.apply($, fetchRequests);
```
Comment (epicfaace) [inline frontend/src/components/worksheets/BundleDetail/BundleDetail.jsx]:
```suggestion
                await $.when.apply($, fetchRequests);

```

File: frontend/src/components/worksheets/NewUpload/NewUpload.jsx [hunk 2]
```diff
@@ -193,68 +193,67 @@ class NewUpload extends React.Component<{
...
-                                    jqHXR.responseText,
-                                    'refresh and try again.',
-                                ),
-                            );
-                            this.props.onUploadFinish();
+            });
+    
+        } catch (error) {
+            this.clearProgress();
+            alert(createAlertText(url, error.responseText));
```
Comment (epicfaace) [inline frontend/src/components/worksheets/NewUpload/NewUpload.jsx:227]: Shouldn't the `'refresh and try again.'` and `onUploadFinish` be called here? I think you might have switched the two `catch` statements.

File: frontend/src/components/worksheets/WorksheetNameSearch.js [hunk 3]
```diff
@@ -19,8 +19,9 @@ export default class extends React.Component {
         }
 
         try {
-            const response = await fetch(`/rest/worksheets?specs=${name}`).then((e) => e.json());
-            const uuid = response.data[0].id;
+            const response = await fetch(`/rest/worksheets?specs=${name}`);
```
Comment (epicfaace) [inline frontend/src/components/worksheets/WorksheetNameSearch.js]: I think in this case, we don't necessarily need to add the additional `await` here. Personally, `.then(e => e.json())` seems simpler / more readable than having to declare a new variable called `json`.

File: frontend/src/serviceWorker.js [hunk 4]
```diff
@@ -29,7 +29,7 @@ export function register(config) {
             return;
         }
 
-        window.addEventListener('load', () => {
+        window.addEventListener('load', async () => {
```
Comment (epicfaace) [inline frontend/src/serviceWorker.js]: I think this file is generated by create-react-app, so we shouldn't have to modify it -- let's leave it as is.

File: frontend/src/util/cli_utils.js [hunk 5]
```diff
@@ -24,17 +24,15 @@ import $ from 'jquery';
...
-    })
-    .always(function() {
-        terminal.resume();
-        if (!isEnabled) {
-            terminal.disable();
-        }
-        self.props.reloadWorksheet();
-    });
+    } finally {
+    terminal.resume();
```
Comment (epicfaace) [inline frontend/src/util/cli_utils.js]: add indent here

File: frontend/src/components/worksheets/NewUpload/NewUpload.jsx (hunk 2 above)
Comment (Levilian) [reply frontend/src/components/worksheets/NewUpload/NewUpload.jsx:227]: No - here we're simply creating a bundle. Sending the data comes after the successful creation of the bundle.

File: frontend/src/components/Bundle/Bundle.js [hunk 6]
```diff
@@ -148,9 +148,8 @@ class Bundle extends React.Component<
                             }
                         }.bind(this),
                     );
-                    $.when.apply($, fetchRequests).then(() => {
```
Comment (epicfaace) [inline frontend/src/components/Bundle/Bundle.js:151]: I'm actually a bit confused about what this logic is actually doing. Why is it calling `$.when` with `fetchRequests` as the argument twice?

File: frontend/src/components/worksheets/BundleDetail/BundleDetail.jsx [hunk 7]
```diff
@@ -174,9 +174,8 @@ class BundleDetail extends React.Component<
                         }
                     }.bind(this),
                 );
-                $.when.apply($, fetchRequests).then(() => {
```
Comment (epicfaace) [inline frontend/src/components/worksheets/BundleDetail/BundleDetail.jsx:177]: Similar question here

File: frontend/src/components/Bundle/Bundle.js (hunk 6 above)
Comment (Levilian) [reply frontend/src/components/Bundle/Bundle.js:151]: https://medium.com/sungthecoder/making-multiple-ajax-calls-and-deciphering-when-apply-array-b35d1b4b1f50
So here we're doing multiple AJAX calls and only go on to the then command if all of them are resolved.

File: frontend/src/components/worksheets/BundleDetail/BundleDetail.jsx (hunk 7 above)
Comment (Levilian) [reply frontend/src/components/worksheets/BundleDetail/BundleDetail.jsx:177]: https://medium.com/sungthecoder/making-multiple-ajax-calls-and-deciphering-when-apply-array-b35d1b4b1f50
So here we're doing multiple AJAX calls and only go on to the then command if all of them are resolved.

File: frontend/src/components/Bundle/Bundle.js (hunk 6 above)
Comment (epicfaace) [reply frontend/src/components/Bundle/Bundle.js:151]: Ah, I see. I think it's clearer to instead just do `Promise.all(...fetchRequests)` as that is native to JS and doesn't require jQuery.

I'm still confused as to why we also have `$.when(fetchRequests)` on line 154, though? That's what I meant about "calling $.when with fetchRequests as the argument twice".

Feel free to ping me on Slack so we can discuss this further.

**Review by epicfaace — APPROVED** [review]

**Pull request merged**